librad/src/net/peer.rs

```diff
@@ -503,13 +520,29 @@ where
 {
     type Update = Gossip;
 
-    async fn put(&self, provider: PeerId, has: Self::Update) -> PutResult {
+    async fn put(&self, provider: PeerId, mut has: Self::Update) -> PutResult<Self::Update> {
         let span = tracing::info_span!("Peer::LocalStorage::put");
         let _guard = span.enter();
 
+        let git = self.inner.get().await.unwrap();
```

Comment:
You need to create a block here -- otherwise the lock is held for the scope of this method

Reply:
Good shout :+1: